projects/igniteui-angular/src/lib/grids/state-base.directive.ts

```diff
@@ -429,16 +431,6 @@ export class IgxGridStateBaseDirective {
         }
     };
 
-    /**
-     *  Event emitted when set state is called with a string.
-     * Returns the parsed state object so that it can be further modified before applying to the grid.
-     * ```typescript
-     * this.state.stateParsed.subscribe(parsedState => parsedState.sorting.forEach(x => x.strategy = NoopSortingStrategy.instance()});
-     * ```
-     */
-    @Output()
-    public stateParsed = new EventEmitter<IGridState>();
```

Comment:
I am not sure why the move the event from base directive.
But anyway the event is emitted in `setStateInternal`, so emit should be moved outside in the inheriting directive and component as well.

Reply:
@skrustev can you please review it again?